hazelcast/src/main/java/com/hazelcast/instance/DefaultAddressPicker.java

```diff
@@ -66,9 +75,9 @@
     private Address publicAddress;
     private Address bindAddress;
 
-    DefaultAddressPicker(Config config, HazelcastProperties hazelcastProperties, ILogger logger) {
+    DefaultAddressPicker(Config config, ILogger logger) {
         this.config = config;
-        this.hazelcastProperties = hazelcastProperties;
+        this.hazelcastProperties = new HazelcastProperties(config);
```

Comment:
What is the advantage of creating another HazelcastProperties? It makes debugging more complex and increases memory usage since the content of the config is copied.

Reply:
Advantage is to pass just `config` to constructor instead of both `config` and `hazelcastProperties`.  I don't think it has significant memory overhead.